core/src/main/java/hudson/util/io/ParserConfigurator.java

```diff
@@ -73,22 +74,37 @@ public void configure(SAXReader reader, Object context) {}
     }
 
     public static void applyConfiguration(SAXReader reader, Object context) throws IOException, InterruptedException {
-        Collection<ParserConfigurator> all = Collections.emptyList();
+        Collection<ParserConfigurator> all;
 
-        if (Jenkins.getInstanceOrNull()==null) {
-            Channel ch = Channel.current();
-            if (ch!=null)
-                all = ch.call(new SlaveToMasterCallable<Collection<ParserConfigurator>, IOException>() {
-
-                    private static final long serialVersionUID = -2178106894481500733L;
-
-                    public Collection<ParserConfigurator> call() throws IOException {
-                        return new ArrayList<ParserConfigurator>(all());
-                    }
-                });
+        if (Jenkins.getInstanceOrNull() == null) {
+            all = fromCacheOrRemote();
         } else
             all = all();
+
         for (ParserConfigurator pc : all)
             pc.configure(reader,context);
     }
+
+    private static Collection<ParserConfigurator> fromCacheOrRemote() throws IOException, InterruptedException {
+        if (cache != null) {
+            return cache;
```

Comment:
You can use `ExtensionListListener` on the master side, but then how would you notify the agent to expire its cache? Could become quite tricky.